# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cache/DBSchema.cpp
# Commit: 0e68db4ad6af
# Full Hash: 0e68db4ad6afe1f43d97533316d80e5aadd96082
# Author: Luca Greco <lgreco@mozilla.com>
# Date: 2020-07-01 21:32:12
# Regressor Bug: 1609920
# File Overlap Count: 2
# Description:
#   Bug 1609920 - part 1: Allow the WebExtension Framework to register a moz-extension service worker. r=dom-workers-and-storage-reviewers,asuth,mixedpuppy
#   
#   - Adds the new about:config pref "extensions.backgroundServiceWorker.enabled" (currently defaults to false).
#   - Adds the background.service_worker property to the manifest JSON schema definition
#   - Locks background.service_worker manifest property behind the new preference
# ==============================================================================

diff -r 2a107eabe00c -r 0e68db4ad6af dom/cache/DBSchema.cpp
--- a/dom/cache/DBSchema.cpp	Mon Jun 29 16:18:44 2020 +0000
+++ b/dom/cache/DBSchema.cpp	Wed Jun 10 12:13:29 2020 +0000
@@ -18,6 +18,7 @@
 #include "mozilla/dom/cache/Types.h"
 #include "mozilla/dom/cache/TypeUtils.h"
 #include "mozilla/net/MozURL.h"
+#include "mozilla/StaticPrefs_extensions.h"
 #include "mozIStorageConnection.h"
 #include "mozIStorageStatement.h"
 #include "mozStorageHelper.h"
@@ -2489,7 +2490,10 @@
 
 #ifdef DEBUG
     nsDependentCSubstring scheme = url->Scheme();
-    MOZ_ASSERT(scheme == "http" || scheme == "https" || scheme == "file");
+    MOZ_ASSERT(
+        scheme == "http" || scheme == "https" || scheme == "file" ||
+        (StaticPrefs::extensions_backgroundServiceWorker_enabled_AtStartup() &&
+         scheme == "moz-extension"));
 #endif
 
     nsCString origin;